pandas/io/stata.py

```diff
@@ -1063,7 +1027,9 @@ def __init__(
         self._column_selector_set = False
         self._value_labels_read = False
         self._data_read = False
-        self._dtype = None
+        # TODO: Place holder for mypy, is this right?
+        self._dtype = np.dtype("int")
+        self._dtype_setup = False
```

Comment:
I don't think you need to add this complexity. where you get a mypy error about self._dtype being None in methods that are only called after self._dtype has been initialized, you can just add `assert self._dtype is not None` at the start of the method and mypy will narrow the types and pass.

Reply:
Thanks.  Definitely better.

Reply:
for posterity.. 
see https://mypy.readthedocs.io/en/latest/kinds_of_types.html#optional-types-and-the-none-type

"Sometimes mypy doesn’t realize that a value is never None. This notably happens when a class instance can exist in a partially defined state, where some attribute is initialized to None during object construction, but a method assumes that the attribute is no longer None. Mypy will complain about the possible None value. You can use assert x is not None to work around this in the method"